Pivot Grid value config type does not match the PivotAggregation type.

## Description  
The `PivotAggregation` type is defined as `(members: any[], data: any[]) => any;` while the `IPivotAggregator` property `aggregator` is defined as `(members: any[], data?: any[]) => any;`, which does not match the `PivotAggregation` type.

 * igniteui-angular version: any
 * browser: any

## Steps to reproduce  

1. Install the attached sample.
2. In the `tsconfig.json` add:
```
"strict": true,
"noImplicitAny": false,
```
4. Run the sample
5. Observe the errors

## Result  
The following errors are thrown:

```
Error: src/app/pivot-grid/pivot-features/pivot-features.component.ts:115:21 - error TS2322: Type 'PivotAggregation' is not assignable to type '(members: any[], data?: any[] | undefined) => any'.

115                     aggregator: IgxTotalSaleAggregate.totalMax,
                        ~~~~~~~~~~

  node_modules/igniteui-angular/lib/grids/pivot-grid/pivot-grid.interface.d.ts:70:5
    70     aggregator?: (members: any[], data?: any[]) => any;
           ~~~~~~~~~~
    The expected type comes from property 'aggregator' which is declared here on type 'IPivotAggregator'
```

## Expected result  
There should be no errors.


## Attachments
[pwrfvl--run.zip](https://github.com/user-attachments/files/18026436/pwrfvl--run.zip)

--- a/projects/igniteui-angular/grids/core/src/pivot-grid.interface.ts
+++ b/projects/igniteui-angular/grids/core/src/pivot-grid.interface.ts
@@ -50,7 +50,8 @@ export interface IPivotDimensionStrategy {
 
 /**
 * Interface describing a PivotAggregation function.
-* Accepts an array of extracted data members and a array of the original data records.
+* Accepts an array of extracted data members and an optional array of the original data records,
+* which can be omitted or undefined.
 */
 export type PivotAggregation = (members: any[], data?: any[]) => any;
 

--- a/projects/igniteui-angular/grids/core/src/pivot-grid.interface.ts
+++ b/projects/igniteui-angular/grids/core/src/pivot-grid.interface.ts
@@ -52,7 +52,7 @@ export interface IPivotDimensionStrategy {
 * Interface describing a PivotAggregation function.
 * Accepts an array of extracted data members and a array of the original data records.
 */
-export type PivotAggregation = (members: any[], data: any[]) => any;
+export type PivotAggregation = (members: any[], data?: any[]) => any;
 
 /* marshalByValue */
 /**
@@ -76,7 +76,7 @@ export interface IPivotAggregator {
      * Aggregator function can be a custom implementation of `PivotAggregation`, or
      * use predefined ones from `IgxPivotAggregate` and its variants.
      */
-    aggregator?: (members: any[], data?: any[]) => any;
+    aggregator?: PivotAggregation;
 }
 
 /* marshalByValue */

